search.go

```diff
@@ -132,6 +146,18 @@ func newSearchFilterFromQuery(query url.Values) (*packages.Filter, error) {
 	return &filter, nil
 }
 
+func getSpecVersion(version string) (*semver.Version, error) {
+	// version must cointain just <major.minor>
+	if len(strings.Split(version, ".")) != 2 {
+		return nil, fmt.Errorf("invalid version '%s': it should be <major.version>", version)
+	}
+	specVersion, err := semver.NewVersion(version)
+	if err != nil {
+		return nil, fmt.Errorf("invalid spec version '%s': %w", version, err)
+	}
+	return specVersion, nil
+}
```

Comment:
Ensure that versions in query parameters are as expected. That means with the pattern `major.minor`